### ISSUE TITLE
In Safari, when splitting by node_modules, Can't find variable: webpackJsonp

### ISSUE THREAD

**Issue by tremby** [BohdanTkachenko/webpack-split-by-path#13]
I'm using code very much like that suggested in the readme to split `node_modules` into a vendors file and the rest into an app file.

In Safari I'm seeing the error `ReferenceError: Can't find variable: webpackJsonp` when the page loads.

Here's the config:

``` js
var SplitByPathPlugin, config, dev, path, webpack;

path = require('path');

SplitByPathPlugin = require('webpack-split-by-path');

webpack = require('webpack');

require('dotenv').config({
  silent: true
});

dev = (process.env.NODE_ENV || 'development') === 'development';

config = {
  entry: {
    app: path.resolve(__dirname, 'src', 'javascripts', 'main.jsx')
  },
  output: {
    path: path.resolve(__dirname, 'public', 'javascripts'),
    filename: '[name].js',
    chunkFilename: '[name].js'
  },
  module: {
    loaders: [
      {
        test: /\.jsx?$/,
        loader: 'babel',
        query: {
          presets: ['es2015', 'react']
        }
      }, {
        test: /\.css$/,
        loader: 'style!css'
      }, {
        test: /\.scss$/,
        loader: 'style!css!sass'
      }
    ]
  },
  plugins: [
    new SplitByPathPlugin([
      {
        name: 'vendors',
        path: path.join(__dirname, 'node_modules')
      }
    ])
  ]
};

if (dev) {
  config.devtool = 'source-map';
} else {
  config.plugins.push(new webpack.optimize.UglifyJsPlugin({
    minimize: true
  }));
}

module.exports = config;
```

When googling for this error I saw some results which talk about the Commons plugin, but not about this one. The only solutions appeared to be to remove the Commons plugin.

Is this a bug, or do I have something wrong with my configuration?

**Comment by bZichett** [comment]
Have you checked your build files for the webpack bootstrap (that defines webpackJsonp) ?  I had a similar issue and found your thread, but this error was resolved after changing the order with which the files are loaded.  Since you should only have two files, this most probably isn't the problem, but worth trying to switch the order if you haven't

---

Anyway, after that I was getting a different issue: `Uncaught TypeError: Cannot read property 'call' of undefined` using this plugin now (without CommonChunks,) but resolved by pushing the entry point to loaded last, with all of the split-by-named files sandwiched between the bootstrap and the entry point.

**Comment by tremby** [comment]
> Have you checked your build files for the webpack bootstrap (that defines webpackJsonp) ? I had a similar issue and found your thread, but this error was resolved after changing the order with which the files are loaded. Since you should only have two files, this most probably isn't the problem, but worth trying to switch the order if you haven't

`vendors.js` is loaded in the HTML before `app.js`. `vendors.js` starts with the `webpackBootstrap`, which looks to include `webpackJsonp`. `app.js` starts with a call to `webpackJsonp`.

> Anyway, after that I was getting a different issue: Uncaught TypeError: Cannot read property 'call' of undefined using this plugin now (without CommonChunks,) but resolved by pushing the entry point to loaded last, with all of the split-by-named files sandwiched between the bootstrap and the entry point.

I'm not sure what you're suggesting here exactly. Are you just saying move the `<script>` tag for the entry point to last in the HTML? It's already like that.

**Comment by bZichett** [comment]
> vendors.js is loaded in the HTML before app.js. vendors.js starts with the webpackBootstrap, which looks to include webpackJsonp. app.js starts with a call to webpackJsonp.

Would you mind posting the bootstrap section here or in a gist, and I can compare it to the output that I have post webpack-split-by-path?  I'm not entirely sure what you mean by "looks to include".  Does that mean the function definition isn't there like the following?

``` javascript
/******/ (function(modules) { // webpackBootstrap
/******/    // install a JSONP callback for chunk loading
/******/    var parentJsonpFunction = window["webpackJsonp"];
/******/    window["webpackJsonp"] = function webpackJsonpCallback(chunkIds, moreModules) {
...
```

> I'm not sure what you're suggesting here exactly. Are you just saying move the script tag for the entry point to last in the HTML? It's already like that.  

Yes, I have the script tags auto-generating based on whether the server should fetch from a web-pack-dev server or pull from a build directory so I accidentally had the order mixed up due to storing module names / paths in a JSON file.  And to clarify, I only posted that bottom section in case any one else runs into that issue after solving the one you are having.

And by the way, not sure if this is just a toy case you posted, but you probably don't need this plugin if you just want to split vendor code from a single entry.  The Commons Plugin should work just fine there, unless you rather not explicitly define the vendor files you are using ( which I'd understand... )

**Comment by tremby** [comment]
Sure, the first chunk of the vendors.js file is here: http://sprunge.us/UbIT?js

**Comment by bZichett** [comment]
Definitely interesting.  It's identical to mine.  Perhaps you can ensure that file is running by adding a console.log statement right after line 25 (after webpackJsonp func definition is attached to the window) or adding a watch point there in browser.

**Comment by tremby** [comment]
And you're testing on Safari? This is all fine on other browsers for me. I actually can't test on Safari from home; I can test what you suggest once back at a Mac on tuesday.

**Comment by MarkusKramer** [comment]
I believe I'm having a the same problem, but it's not related to safari. I have a "main" entry chunk and  this config

```
config.plugins.push(new SplitByPathPlugin([
    { name: 'vendor', path: path.resolve(__dirname, 'node_modules')}
]));
```

The produced vendor.js as well as main.js start with 
`webpackJsonp(...`
which causes the "Can't find variable: webpackJsonp" for me too.

**Comment by MarkusKramer** [comment]
Solved for me. I needed to add the "manifest" chunk:

```
    <script src="manifest.js"></script>
    <script src="vendor.js"></script>
    <script src="main.js"></script>
```

**Comment by tremby** [comment]
I don't have any such `manifest.js` chunk; where does that come from?

But also this issue was a long time ago and I don't appear to be having the problem any more, even though I am using the split by path plugin. I'm not sure what changed.

**Comment by davelaflam** [comment]
Looks like this isn't a specific browser issue rather a global bug that has come up in the v0.1.0 release. 

If I use "webpack-split-by-path": "0.0.10" then the error does not appear and everything works as expected.

However, if I upgrade to the latest release, "webpack-split-by-path": "0.1.0", then I get the following error. `Uncaught TypeError: Cannot read property 'call' of undefined`

Please let me know if you need any additional information.

**Comment by xpepermint** [comment]
Another victim here :(. I don't use this package, just a webpack and I have a similar issue.

``` js
// a file
"use strict";

const typeable_1 = require('typeable'); // typeable_1 is NOT undefined

function presence(value) {
    return typeable_1.isPresent(value); // typeable_1 is undefined
}
exports.presence = presence;
```

I wonder how this is possible ...

**Comment by BohdanTkachenko** [comment]
We just released 2.0.0 version of this plugin for webpack 2. 
For webpack 1 there is still a branch webpack1 and version 1.0.0 on NPM.
Please consider upgrading to webpack 2 and version 2.0.0 of this plugin.
If this issue still reproduces on 2.0.0 please re-open this issue.
If you really need to have this in 1.0.0, PR's to webpack1 branch are always welcome, but I don't longer plan to fix issues for webpack1.

**Comment by ArmanNisch** [comment]
Hi,

What solved this issue for me was removing the **async** attribute from within the <script> tags:

Remove the **async** attribute:
```
<script async type="text/javascript" src="javascripts/runtime.js"></script>
<script async type="text/javascript" src="javascripts/vendor.js"></script>
<script async type="text/javascript" src="javascripts/app.js"></script>
```
